core/native/src/internal/TimeZoneRules.kt

```diff
@@ -0,0 +1,388 @@
+/*
+ * Copyright 2019-2023 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.internal
+
+import kotlinx.datetime.*
+import kotlin.math.*
+
+internal sealed interface OffsetInfo
```

Comment:
This file has too many classes. Could you extract hierarchies like `OffsetInto` and `DateOfYear` to separate files?
Also consider using nested classes, because otherwise it is unclear, for example, where `kotlinx.datetime.internal.Gap` belongs to.